workflow-testing/src/main/java/com/squareup/workflow1/testing/WorkflowTestRuntime.kt

```diff
@@ -435,9 +435,9 @@ private fun <T> unwrapCancellationCause(block: () -> T): T {
   } catch (e: CancellationException) {
     throw generateSequence(e as Throwable) { e.cause }
         // Stop the sequence if an exception's cause is itself.
-        .scanReduce { error, cause ->
+        .runningReduce { error, cause ->
```

Comment:
Nit: for the same behavior you can utilize `windowed` function, and it'll be more straightforward for a reader. I won't  guarantee it'll fit well with what you're trying to achieve in the next call, but anyway this part could benefit from some rethink of conditions and tinkering with stdlib functions.

At the same time, no objections in merging it as is.

Reply:
I don't see `windowed` in the docs – what am I missing?

Reply:
Yea, we'd need another operator to reduce the emitted lists I believe.

Reply:
Either https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/zip-with-next.html or https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/windowed.html could be a good candidate, but I don't know how will it help with the methods down the chain.

Reply:
Yea, both those would require another operator to flatten the lists they return. `runningReduce` does it in one. I'm not sure which is more readable.